packages/nc-gui/components/smartsheet/Grid.vue

```diff
@@ -56,8 +56,11 @@ const hasEditPermission = isUIAllowed('xcDatatableEditable')
 const isView = false
 
 const selected = reactive<{ row: number | null; col: number | null }>({ row: null, col: null })
+const selectedRows = reactive<{ startcol: number | null; endcol: number | null; startrow: number | null; endrow: number | null }>({ startcol: null, endcol: null, startrow: null, endrow: null })
+const rangerows = reactive<{ minrow: number | null; maxrow: number | null; mincol: number | null; maxcol: number | null }>({ mincol: null, maxcol: null, minrow: null, maxrow: null })
```

Comment:
- Why use `number | null` instead of just `number` and assign `NaN` as a falsy value and test against that?
That would make the Generic of `reactive` unnecessary and you could shorten this to
```ts
const selectedRows = reactive({ startcol: NaN, endcol: NaN, startrow: NaN, endrow: NaN })
```

- Use proper pascal casing, i.e. `startcol` should be `startCol`